rollup/fees/rollup_fee.go

```diff
@@ -210,12 +214,21 @@ func calculateEncodedL1DataFeeFeynman(
 	execScalar *big.Int,
 	blobScalar *big.Int,
 ) *big.Int {
-	// tx size (RLP-encoded)
 	txSize := big.NewInt(int64(len(data)))
 
-	// compression_ratio(tx) = 1 (placeholder, scaled to match scalars precision)
+	// Default compression ratio is 1.0 (no compression)
 	compressionRatio := big.NewInt(rcfg.Precision.Int64())
 
+	compressedBytes, err := zstd.CompressScrollBatchBytes(data)
```

Comment:
might still be good to use the zstd version that comes with da-codec

Reply:
Good idea. However, the tx hash is not available here; logged the raw data instead (whole info of the function input, for debugging): [f31326d](https://github.com/scroll-tech/go-ethereum/pull/1197/commits/f31326debd6a5f65d485e2949c837ddb95ba7586). And if it's also very helpful to add tx hash as well, we can add a tx hash param in `calculateEncodedL1DataFeeFeynman`.

Reply:
- might still be good to use the zstd version that comes with da-codec

Reply:
I don't understand your point about the counter-intuitive call stack.

But I agree we should:
1. Pass `compressionRatio` as parameter for easier testing, and
2. Expose an estimate tx compression ratio function in `da-codec`

Reply:
Replied to the comment here: https://github.com/scroll-tech/go-ethereum/pull/1196#discussion_r2166434151.